src/service/user-service.js

```diff
@@ -0,0 +1,138 @@
+import { v4 } from 'uuid'
+import bcrypt, { compare } from 'bcrypt'
+import TokenService from './token-service.js'
+import ApiError from '../exceptions/api-error.js'
+import MailService from './mail-service.js'
+import { appAddresses, salt } from '../data/config.js'
+import User from '../models/User.js'
+
+const { backend } = appAddresses
+
+class UserService {
+  async signUpUser(name, email, password, mobileNumber) {
+    const userExistsByEmail = await User.findOne({ email })
+    const userExistsByPhone = await User.findOne({ mobileNumber })
+
+    if (userExistsByEmail || userExistsByPhone) {
+      throw ApiError.BadRequest('User with this email or phone already exists')
+    }
+
+    const quantitySalt = await bcrypt.genSalt(Number(salt))
+    const hashPassword = await bcrypt.hash(password, quantitySalt)
+
+    const activationLink = v4()
+    const fullActivationLink = `${backend}/api/activate/${activationLink}`
+
+    const newUser = await User.create({
+      name,
+      email,
+      password: hashPassword,
+      mobileNumber,
+      loginAttempts: 0,
+      lockUntil: null,
+      activationLink,
+    })
+
+    await MailService.sendActivationMail(email, fullActivationLink)
+
+    const { accessToken, refreshToken } = await TokenService.generateTokens(
+      newUser._id,
+      newUser.email,
+    )
+
+    await TokenService.saveRefreshToken(newUser._id, refreshToken)
+
+    return { accessToken, refreshToken, user: newUser }
+  }
+
+  async activate(activationLink) {
+    const user = await User.findOne({ activationLink })
+
+    if (!user) {
+      throw ApiError.BadRequest('Incorrect activation link')
+    }
+
+    user.isActivated = true
+    await user.save()
+  }
+
+  async signInUser(email, password) {
```

Comment:
It seems that this method violates the SRP, that's why you need to create different functions for different responsibilities. Please consider creating separate methods:

- getUserByEmail
- handleInvalidUser
- handleIncorrectPassword
- resetLockAndLoginAttempts